sql/src/main/java/org/apache/druid/sql/calcite/expression/DefaultOperandTypeChecker.java

```diff
@@ -149,7 +147,25 @@ public SqlOperandCountRange getOperandCountRange()
   @Override
   public String getAllowedSignatures(SqlOperator op, String opName)
   {
-    return SqlUtil.getAliasedSignature(op, opName, operandTypes);
+    final List<?> operands = !operandNames.isEmpty() ? operandNames : operandTypes;
+    final StringBuilder ret = new StringBuilder();
+    ret.append("'");
+    ret.append(opName);
+    ret.append("(");
+    for (int i = 0; i < operands.size(); i++) {
+      if (i > 0) {
+        ret.append(", ");
+      }
+      if (i >= requiredOperands) {
+        ret.append("[");
+      }
+      ret.append("<").append(operands.get(i)).append(">");
+    }
+    for (int i = requiredOperands; i < operands.size(); i++) {
+      ret.append("]");
+    }
```

Comment:
this doesn't seem right. is there a test case to verify signature when there are multiple required operands?

Reply:
Hmm, git diff shows that I moved `BasicOperandTypeChecker` -> `DefaultOperandTypeChecker`. However `DefaultOperandTypeChecker` was already present in `OperatorConversions` that I pulled out.

I think it creates a signature like:
```
functionName(requiredOperand1, requiredOperand2, [optionalOperand1, [optionalOperand2]])
```

Reply:
There's an `APPROX_COUNT_DISTINCT_DS_HLL` that takes 3 parameters, out of which the first one is required and the rest are optional.
There's `BLOOM_FILTER` that takes 2 parameters and both are required
And then there's `APPROX_QUANTILE_FIXED_BUCKETS` which takes in 6 parameters out of which first 5 are required.

Reply:
ah yes. Now I get it. thanks for explaining the flow 😄